### PR TITLE
Month1-Week2-Section2

### PR DESCRIPTION
**Pull request by camodesigns**

### CODE REVIEW

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <string>
+#include <iostream>
+#include <string>
+class FSong
+{
+private:
+    int MinutesDuration;
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h]: Please follow the exercise's instructions closely. You were supposed to just have one variable for the duration which holds the total seconds of the song. It is important to differentiate the way we want to store our data and the way we want to display it. Of course, to display the duration, the format `minutes:seconds`, but it is far less complicated to store the data as just one variable. That way there aren't different values storing the same data.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    int SecondsDuration;
+    std::string ArtistName;
+    std::string SongName;
+public:
+    FSong();
+
+    FSong(int EntryMinutesDuration, int EntrySecondsDuration, std::string EntryArtistName, std::string EntrySongName);
+    std::string GetFormatedDuration()const;
+    bool bIsvalid();
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h]: Remember to use PascalCase and functions don't need to have a b prefix. The documentation has some examples for boolean functions.
image

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <string>
+#include <iostream>
+#include <string>
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h]: You're including string twice

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
+#include "Song.h"
+FSong::FSong()
+{
+    MinutesDuration = 0;
+    SecondsDuration = 0;
+    ArtistName = "";
+    SongName = "";
+}
+FSong::FSong(int EntryMinutesDuration, int EntrySecondsDuration, std::string EntryArtistName, std::string EntrySongName)
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: Remember that parameters should be const if they are not modified inside the function. Please make this change for the whole review

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+{
+    MinutesDuration = EntryMinutesDuration;
+    SecondsDuration = EntrySecondsDuration;
+    ArtistName = EntryArtistName;
+    SongName = EntrySongName;
+
+}
+
+
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: Remember that consistency is important. Here you have 3 spaces between functions, then you have none, then you have one, and then you have none for the rest of the file.

I consider one space to be the most legible and clear.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+std::string FSong::GetFormatedDuration()const
+{
+    std::string FormatedDuration = "";
+    FormatedDuration = "Minutes: " + std::to_string(MinutesDuration) + "\t" "Seconds" + std::to_string(SecondsDuration);
+    return FormatedDuration;
+}
+bool FSong::bIsvalid()
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: This method should be const, since it's not modifying the object in any way :)

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+{
+    if (MinutesDuration <= 0 && SecondsDuration < 0)
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: This code works perfectly fine, but it is a bit complex for how trivial it is. Instead of writing
``` 
if (bCondition)
{
    return true;
}
else
{
    return false;
}
```
you could just write
```
return bCondition;
```
So, in this case, you could just write:
```
return MinutesDuration > 0 && SecondsDuration >= 0;
```

That way your function is clearer and easier to read, and you saved 7 lines for such a simple function.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "Song.h"
+
+class FPlayList {
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: Remember to follow conventions. This curly brace is not following our standards.

Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: Also, "playlist" is just one word hehe, it should be FPlaylist instead of FPlayList

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+private:
+	std::string PlayListTitle;
+	const static int MaxSongCount = 30;
+	int SongCount = 0;
+	FSong Songs[MaxSongCount];
+	
+public:
+	FPlayList();
+	FPlayList(std::string NewPlayListTitle);
+	std::string GetListTitle();
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: Careful with these kinds of abbreviations. `GetTitle()` and `SetTitle(...)` is more than enough since the context already makes it clear that you mean the playlist title. However, if you really want to be specific, writing `List` instead of `Playlist` reduces clarity and makes the name ambiguous, especially considering what I commented before saying that Playlist is just one word.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+	void SetListTitle(std::string NewPlayListTitle);
+	bool bAddSong(FSong FSong);
+	FSong GetSong(int SongIndex);
+	void  PlayListDisplay();
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: Remember that functions should start with a verb since they should describe what they do. This name would be just fine if it was written the other way around. `DisplayPlaylist` is a lot clearer than `PlaylistDisplay`

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.cpp [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+	return false;
+}
+
+
+void  FPlayList::PlayListDisplay()
+{
+	std::cout << "PlayList" << std::endl;
+	std::cout << "Song Title\t\t\t\tArtist Name\t\t\t\tDuration" << std::endl;
+	std::cout << "------------------------------------------------------------------------------------------" << std::endl;
+	for (int CurrentSong = 0; CurrentSong < SongCount; CurrentSong++)
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.cpp]: This is an example of misleading names. I know you like to name your indices explicitly and that's actually very nice. However, keep in mind that they are still _indices_. `CurrentSong` is not actually the current song, but rather the current song _index_. By itself, this can be misleading, but it becomes a bigger problem since now you have used that name, and so you can't use it for anything else.

For that reason, when you actually get the song using the index, you can't name it `CurrentSong` so you had to name it `ListedSong`. Now, by just reading the names, would you know what the difference between `CurrentSong` and `ListedSong` is? If it was me, I'd assume that `CurrentSong` is the actual current song and I would have no idea what `ListedSong` is, which is not good 😅 .

It's cases like this one that show why being explicit is important. `CurrentSong` should instead be `CurrentSongIndex` or something like that, and `ListedSong` should be `CurrentSong`. What do you think?

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.h [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class FPlayListContainer {
+private:
+	const static int MaxPlayListCount = 5;
+	FPlayList PlayLists[MaxPlayListCount];
+	int PlayListCount;
+public:
+	FPlayListContainer();
+	FPlayListContainer(FPlayList NewPlayList);
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.h]: Do you really need a constructor that takes in a playlist for a container of playlists?

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.cpp [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+FPlayListContainer::FPlayListContainer()
+{
+	PlayListCount = 0;
+}
+
+FPlayListContainer::FPlayListContainer(FPlayList NewPlayList)
+{
+	PlayLists[PlayListCount] = NewPlayList;
+	PlayListCount++;
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.cpp]: Careful. PlaylistCount has not been initialized at this point, it could be any random value. PlaylistCount++ is not guaranteed to be 1 every time.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+        while (!std::cin.good() || OptionChoice < 1 || OptionChoice > 3)
+        {
+            std::cin.clear();
+            std::cin.ignore(1000, '\n');
+            std::cout << "Invalid input,Please enter Your new choice or selection" << std::endl;
+            std::cin >> OptionChoice;
+        }
+        ActiveCommand = (MenuCommand)OptionChoice;
+        system("CLS");
+    } while (!std::cin.good());
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Doesn't seem like you need this `do-while` since you already have another while checking for this same condition.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+
+    std::cout << "Creating a new Playlist Fantastic, Please enter a name for Your Playlist: " << std::endl;
+    std::string PlayListName;
+    std::getline(std::cin, PlayListName);
+    while (!std::cin.good() || PlayListName == "" || std::cin.peek() == '\n')
+    {
+        std::cout << "Invalid input, please enter Your playlist name again: " << std::endl;
+        std::getline(std::cin, PlayListName);
+    }
+    FPlayList ANewPlayList(PlayListName);
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Why do you need that `A` prefix? Why not `NewPlaylist`?

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp [hunk 16]
```diff
@@ -0,0 +1,241 @@
...
+        ANewPlayList.bAddSong(NewSong);
+        std::cout << "Song add sucefull, What do you whant to do now?" << std::endl;
+        std::cout << "1. Enter another song" << std::endl;
+        std::cout << "2.Go to main menu" << std::endl;
+        std::cin >> AddSongs;
+      
+        std::cin.clear();
+        std::cin.ignore(1000, '\n');
+        system("CLS");
+    } while (AddSongs == 1);
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: AddSongs is not an acceptable name for a variable that represents the menu choice of the user. 👀

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+    PlayListContainer.AddPlaylist(ANewPlayList);
+    ActiveCommand = MenuCommand::MainMenu;
+}
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp:170]: This method is very nice! But I would suggest using a bit more spaces. It is hard to follow the function since there are hardly any spaces, so you get no breaks when reading through it.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp [hunk 18]
```diff
@@ -0,0 +1,241 @@
...
+        std::getline(std::cin, PlayListName);
+    }
+    FPlayList ANewPlayList(PlayListName);
+    do
+    {
+        system("CLS");
+        std::cout << "Well, now add a song to " << PlayListName << " Playlist" << std::endl;
+        std::cout << "Now,Follow the instructions" << std::endl;
+
+        int EntryMinutesDuration;
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Do you really need to declare all these variables here? Why not declare them closer to where they are actually defined?

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp [hunk 19]
```diff
@@ -0,0 +1,241 @@
...
+
+
+void FSpotify::OpenBrowsePlayListMenu()
+{   
+
+    std::cout << "Here are Your Playlists!" << std::endl;
+    int Choice = 1;
+    do{
+        system("CLS");
+     for (int ListWalker = 0;ListWalker < 5; ListWalker++)
```
Comment (JDAlzate) [inline Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Again, it's nice that you want to make your indices more explicit, but it loses it's purpose if the name is not clearer than `i` 😅 . ListWalker doesn't really tell me anything. Instead, you should make it clear that this is an index, so what about `PlaylistIndex`?

**Review by JDAlzate — CHANGES_REQUESTED** [review]
Nice job! You're app is looking pretty good :) But there are a few things that do need to be changed.

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp (hunk 6 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: Changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp (hunk 7 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: Wow, I didn't know that method, I really thank you

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h (hunk 8 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: all the PlayList changed for Playlist

Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h (hunk 9 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: Modified

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h (hunk 10 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.h]: modified

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.cpp (hunk 11 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.cpp]: It's reasonable, thank you, changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.h (hunk 12 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.h]: You havve reason, not necesary I delete that constructor

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.cpp (hunk 13 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayListContainer.cpp]: I delete that constructor and I change  the PlaylistCount in FPlaylistContainer::AddPlaylist(FPlaylist NewPlaylist) for PlaylistCount +1

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp (hunk 14 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp (hunk 15 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp (hunk 16 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp (hunk 17 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp:170]: Spaces Added

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h (hunk 1 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h]: Changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h (hunk 2 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h]: ok, I understand, changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h (hunk 3 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.h]: OMG, I Look, #include <string>Deleted

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp (hunk 5 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: Changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp (hunk 18 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: reubicated

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp (hunk 19 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp]: Changed

File: Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp (hunk 4 above)
Comment (camodesigns) [reply Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Song.cpp]: Ok, Understanded, and Added in all the program
